cognite/extractorutils/configtools.py

```diff
@@ -243,6 +249,33 @@ def __repr__(self) -> str:
         return self._expression
 
 
+def _load_certificate_data(cert_path: str, password: Optional[str]) -> Tuple[str, str]:
+    path = Path(cert_path)
+    cert_data = Path(path).read_bytes()
+
+    if path.suffix == ".pem":
+        cert = load_pem_x509_certificate(cert_data)
+        private_key = serialization.load_pem_private_key(cert_data, password=password.encode() if password else None)
+        private_key_str = private_key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.TraditionalOpenSSL,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+        return (base64.b16encode(cert.fingerprint(hashes.SHA1())), private_key_str)
+    elif path.suffix == ".pfx":
+        (private_key, cert, _) = pkcs12.load_key_and_certificates(
+            cert_data, password=password.encode() if password else None
+        )
+        private_key_str = private_key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.TraditionalOpenSSL,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+        return (base64.b16encode(cert.fingerprint(hashes.SHA1())), private_key_str)
+    else:
+        raise InvalidConfigError("Unknown certificate format")
```

Comment:
Nit: maybe include what we think the format is, and what formats are allowed. This exception message is typically shown to the user, so it could help figuring out what's wrong

``` suggestion
        raise InvalidConfigError(f"Unknown certificate format '{path.suffix}'. Allowed formats are 'pem' and 'pfx'")
```